csv/result.go

```diff
@@ -602,7 +612,11 @@ func (d *tableDecoder) init(line []string) error {
 	}
 
 	key := execute.NewGroupKey(keyCols, keyValues)
-	d.builder = execute.NewColListTableBuilder(key, newUnlimitedAllocator())
+	alloc := d.c.Allocator
+	if alloc == nil {
+		alloc = &memory.Allocator{}
```

Comment:
It's a bit spotty at the moment, but it's mostly the arrow functions that take that. I'm not sure I would rely on that especially since `ColListTableBuilder` uses the allocator in weird ways that nobody else does.